packages/paste-core/components/menu/src/MenuItem.tsx

```diff
@@ -85,23 +107,90 @@ export const StyledMenuItem = React.forwardRef<HTMLDivElement | HTMLAnchorElemen
         }}
         ref={ref}
       >
-        {children}
+        {props['aria-checked'] === true ? (
+          <Box as="span" display="flex" justifyContent="space-between" alignItems="center">
+            {children}
+            <SelectedIcon decorative color={getCheckedIconColor({disabled: props.disabled, variant})} />
+          </Box>
+        ) : (
+          children
+        )}
       </Box>
     );
   }
 );
 
 StyledMenuItem.displayName = 'StyledMenuItem';
 
+const getComputedVariant = (variant: MenuItemVariant, isGrouped: boolean): MenuItemVariant => {
+  if (isGrouped) {
+    if (variant === MenuItemVariants.DESTRUCTIVE) {
+      return MenuItemVariants.DESTRUCTIVE_GROUP_ITEM;
+    }
+    return MenuItemVariants.GROUP_ITEM;
+  }
+  return variant;
+};
+
 const MenuItem = React.forwardRef<HTMLDivElement, MenuItemProps>(
   ({as = StyledMenuItem, variant: _variant, element = 'MENU_ITEM', ...props}, ref) => {
-    let variant = _variant;
     const isGrouped = React.useContext(MenuGroupContext) === MenuItemVariants.GROUP_ITEM;
-    if (isGrouped && _variant === MenuItemVariants.DESTRUCTIVE) variant = MenuItemVariants.DESTRUCTIVE_GROUP_ITEM;
-    else if (isGrouped) variant = MenuItemVariants.GROUP_ITEM;
+    const variant = _variant ? getComputedVariant(_variant, isGrouped) : _variant;
 
     return <MenuPrimitiveItem {...props} element={element} variant={variant} as={as} ref={ref} />;
   }
 );
 MenuItem.displayName = 'MenuItem';
+MenuItem.propTypes = {
+  href: PropTypes.string,
+  variant: PropTypes.oneOf([Object.values(MenuItemVariants)]),
+  disabled: PropTypes.bool,
+  id: PropTypes.string,
+  onClick: PropTypes.func,
+  element: PropTypes.string,
+};
 export {MenuItem};
+
+const MenuItemCheckbox = React.forwardRef<HTMLDivElement, MenuItemCheckboxProps>(
+  ({as = StyledMenuItem, variant: _variant, element = 'MENU_ITEM_CHECKBOX', ...props}, ref) => {
+    const isGrouped = React.useContext(MenuGroupContext) === MenuItemVariants.GROUP_ITEM;
+    const variant = _variant ? getComputedVariant(_variant, isGrouped) : _variant;
+
+    return <MenuPrimitiveItemCheckbox {...props} element={element} variant={variant} as={as} ref={ref} />;
+  }
+);
+MenuItemCheckbox.displayName = 'MenuItemCheckbox';
+MenuItemCheckbox.propTypes = {
+  href: PropTypes.string,
+  variant: PropTypes.oneOf([Object.values(MenuItemVariants)]),
+  disabled: PropTypes.bool,
+  id: PropTypes.string,
+  onClick: PropTypes.func,
+  element: PropTypes.string,
+  value: PropTypes.string,
+  checked: PropTypes.bool,
+  name: PropTypes.string,
+};
+export {MenuItemCheckbox};
+
+const MenuItemRadio = React.forwardRef<HTMLDivElement, MenuItemRadioProps>(
```

Comment:
suggestion (not-blocking): kinda feel like some of the things in this file can be cleanly pulled out into their own files

Reply:
File per component probably, it is getting a little messy